enhancements/external-secrets-operator/external-secrets-component-config.md

```diff
@@ -0,0 +1,321 @@
+---
+title: external-secrets-component-config
+authors:
+  - "@sbhor"
+reviewers:
+  - "@tgeer"
+approvers:
+  - "@tgeer"
+api-approvers:
+  - "@tgeer"
+creation-date: 2025-12-1
+last-updated: 2025-12-5
+tracking-link: 
+  - https://issues.redhat.com/browse/ESO-266
+see-also:
+  - NA
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+# Component Configuration for external-secrets Operator
+
+## Summary
+
+This document proposes an enhancement to the `ExternalSecretsConfig` API by introducing a `ComponentConfig` extension and global annotations support through `Annotations` field. This allows administrators to specify component-specific overrides for deployment lifecycle settings, custom environment variables, and global custom annotations for external-secrets components (Controller, Webhook, CertController, BitwardenSDKServer). This change offers administrators greater control over the resource management and operational parameters of components.
+
+## Motivation
+
+Administrators often need to control deployment lifecycle settings and add custom annotations without directly modifying the underlying operator-managed Deployment resources.
+
+### User Stories
+
+- As an administrator, I want to customize deployment lifecycle properties for `external-secrets` operand components to manage their resource consumption and rollback behavior.
+- As an `external-secrets` user, I want to be able to set specific deployment override values independently for each component via ComponentConfig to meet unique operational requirements.
+- As a platform engineer, I want to add custom annotations to external-secrets deployments without modifying operator-managed resources.
```

Comment:
```suggestion
- As an OpenShift Administrator, I want to define custom metadata (like annotations or labels) on the external-secrets component deployments via the `ExternalSecretsConfig` API so that the deployments correctly integrate with cluster policy tools, monitoring systems (e.g., Prometheus), and internal tooling without being overwritten.
```